docs/theory.rst

```diff
@@ -43,29 +43,244 @@ By collecting :math:`x_n` into a complex number array (called :cpp:member:`Qrack
+
+Mechanically, this performs the following mathematical operations:
+
+.. math::
+    :label: x_3bit_opt
 
-This is again an "embarrassingly parallel" operation. Some bits are completely uninvolved and these bits are passed unchanged in each state from input to output. Bits acted on by the register operation have a one-to-one mapping between input and states. This can all be handled through transformation via bit masks on the input state permutation index. Register-like methods generally all use bitmasks to separate involved bits from uninvolved bits, transform the involved bits like above, and use the bit transformation to map state vector coefficients to new positions in the vector, in a unitary manner. (Note that the operation amounts to swapping coefficients in correspondence with the bitmask transforms, not directly acting bitwise operations on the raw representation of the state itself. The state is represented as a set of double precision complex coeffecients, not bits.)
+    \begin{bmatrix}
+      {
+        \begin{bmatrix}
+          0 & 1
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{000} \\
+          x_{001}
+        \end{bmatrix}
+      } \\
+      {
+        \begin{bmatrix}
+          1 & 0
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{000} \\
+          x_{001}
+        \end{bmatrix}
+      } \\
+      {
+        \begin{bmatrix}
+          0 & 1
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{010} \\
+          x_{011}
+        \end{bmatrix}
+      } \\
+      {
+        \begin{bmatrix}
+          1 & 0
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{010} \\
+          x_{011}
+        \end{bmatrix}
+      } \\
+      {
+        \begin{bmatrix}
+          0 & 1
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{100} \\
+          x_{101}
+        \end{bmatrix}
+      } \\
+      {
+        \begin{bmatrix}
+          1 & 0
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{100} \\
+          x_{101}
+        \end{bmatrix}
+      } \\
+      {
+        \begin{bmatrix}
+          0 & 1
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{110} \\
+          x_{111}
+        \end{bmatrix}
+      } \\
+      {
+        \begin{bmatrix}
+          1 & 0
+        \end{bmatrix}
+        \times
+        \begin{bmatrix}
+          x_{110} \\
+          x_{111}
+        \end{bmatrix}
+      }
+    \end{bmatrix}
+    = 
+    \begin{bmatrix}
+      x_{001} \\
+      x_{000} \\
+      x_{011} \\
+      x_{010} \\
+      x_{101} \\
+      x_{100} \\
+      x_{111} \\
+      x_{110}
+    \end{bmatrix}
```

Comment:
This conveys the idea well, but we might want to stipulate that the notation here isn't rigorous. This is a heurism.